armi/reactor/components/componentParameters.py

```diff
@@ -59,20 +59,40 @@ def getComponentParameterDefinitions():
             description="Component temperature in {}".format(units.DEGC),
         )
 
+        pb.defParam(
+            "temperatureInCBOL",
+            units=units.DEGC,
+            description=f"Component temperature in {units.DEGC} at BOL",
+        )
+
         pb.defParam(
             "numberDensities",
             setter=isNumpyArray("numberDensities"),
             units=f"#/(bn*{units.CM})",
             description="Number densities of each nuclide.",
         )
 
+        pb.defParam(
+            "numberDensitiesBOL",
```

Comment:
Honest Question. None blocking to this PR.

This "other parameter at BOL" parameters you created... Could these just be Python variables? Is there some reason they need to be full parameters?

Reply:
hmmmm that is an interesting thought. I bet they could be attributes actually. my only concern would be on a snapshot/restart, but we reload the BOL state when doing restarts/snapshots, so it might be doable actually.

Reply:
For posterity, we found the attribute approach does indeed not work. This is suspected to be due to the fact that while the attributes are reloaded for the blueprints assems at BOL, they are not for the assems in the core. 

So that commit was reverted and to cut down on DB size increase, we only store the heavy metal nuclides + ndens and on components who have heavy metal inventory. The intent here is that this will, in general, be a small(er) subset of the core (e.g., just fuel comps) and not cause unneccessary bloat on the DB.